cfgrammar/src/lib/yacc/info.rs

```diff
@@ -0,0 +1,37 @@
+use crate::yacc::parser::YaccGrammarWarning;
+use crate::yacc::YaccGrammarError;
+
+/// Controls the collection of and collects information about warnings and other information
+/// that result during the construction of a `YaccGrammar`.  If `warnings_as_errors` is `true`
+/// does not collect any warnings.
+pub struct YaccGrammarInfo {
+    pub(crate) warnings_as_errors: bool,
+    warnings: Vec<YaccGrammarWarning>,
+}
+
+impl YaccGrammarInfo {
+    pub fn new(warnings_as_errors: bool) -> Self {
```

Comment:
So, I think it'd probably be better to go for a full builder pattern here, with a `warnings_as_errors().build()` function here.
I was hesitant to just add `warnings_as_errors()` on `YaccGrammarInfo` itself it seems like it would only be confusing if it was callable after construction, since it may not have the desired effect if called too late.

But using the full builder pattern here would solve that, and be easier to add more flags in the future.